File: new-algorithm/src/javaalgo/recursion/boj10826.java
package javaalgo.recursion;

import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;
import java.math.BigInteger;

public class boj10826 {
    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        int n = Integer.parseInt(br.readLine());
        BigInteger[] dp = new BigInteger[10001];

        dp[0] = BigInteger.ZERO;
        dp[1] = BigInteger.ONE;
        for (int i = 2; i <= n; i++) {
            dp[i] = dp[i - 2].add(dp[i - 1]);
        }

        System.out.println(dp[n]);

    }
}

File: new-algorithm/src/kotlinalgo/recursion/boj10826.kt
package kotlinalgo.recursion

import java.io.BufferedReader
import java.io.InputStreamReader
import java.math.BigInteger

fun main() = with(BufferedReader(InputStreamReader(System.`in`))) {
    val n = readLine().toInt()
    val dp = arrayOfNulls<BigInteger>(10001)

    dp[0] = BigInteger.ZERO
    dp[1] = BigInteger.ONE

    for (i in 2..n) {
        dp[i] = dp[i - 2]!!.add(dp[i - 1])
    }
    println(dp[n])

}